### ISSUE TITLE
Support for Edge

### ISSUE THREAD

**Issue by KonTy** [philc/vimium#2076]
Could you guys please consider adding support for MS Edge? All the blogs I see say that most Chrome extensions could be changed support Edge with minimal 1 line changes.

**Comment by adrianmt** [comment]
Yes please.  I like Chrome and Vimium, but Chrome burns through battery faster.
http://bgr.com/2015/10/20/chrome-vs-firefox-battery-life/

**Comment by mihirkumar** [comment]
Yeah guys, I've made the switch to MS Edge after the anniversary update and I would love to have Vimium on it.

**Comment by iscriptology** [comment]
same here... waiting for vimium before switching to MS Edge

**Comment by sergey-rybalkin** [comment]
+1. Vimium is the only thing that keeps me on Chrome

**Comment by russelCrowe** [comment]
Microsoft introduced a tool to convert Chrome extension to an Edge extension.

[Read more here](http://www.winbeta.org/news/microsoft-releases-edge-extension-toolkit-windows-10-helps-devs-port-extensions-chrome)

**Comment by WhiteAbeLincoln** [comment]
Has there been any progress on this? It would be a nice feature.

**Comment by KonTy** [comment]
So no love for Edge? No acknowledgment even fro the devs :(

**Comment by alexsukhodolsky** [comment]
+1

**Comment by KonradJanica** [comment]
I've found this project: https://github.com/iscriptology/edge-vimium

It seems abandoned. I managed to launch it on Edge but there are many bugs. I will try to work on it in my free time here: https://github.com/KonradJanica/edge-vimium

**Comment by KonTy** [comment]
awesome, thanks. I'll see if I can contribute as well. I hate bloated Chrome

**Comment by KonTy** [comment]
Konrad, why fork as oppose to contributing directly to the project? It seems that people keep improving it.

**Comment by KonradJanica** [comment]
I forked it because I assume it is abandoned considering that I have an outstanding pull request for almost 3 weeks now

**Comment by KonTy** [comment]
I see, didn't know that

**Comment by alexsukhodolsky** [comment]
either way you'll need to fork it to make changes. @KonradJanica please open pull requests if you make changes. It'd be nice to have a single plugin to which everyone contributes.

**Comment by smblott-github** [comment]
@KonradJanica... The project is absolutely not abandoned.  I'm just a bit busy with other stuff right now.  There's a fair few of my own PRs sitting in the queue too.  Vimium will get more of my time in a couple of weeks.

**Comment by KonradJanica** [comment]
@smblott-github I was talking about the Edge version of vimium in the repo here: https://github.com/iscriptology/edge-vimium

**Comment by iscriptology** [comment]
@KonradJanica I saw your PR it just wasn't significant enough to merge. Please make a PR once your fork is starting to mature.
blessing,

**Comment by zhouyangnk** [comment]
I really really really wait in hope for Edge's vimium.

**Comment by KonTy** [comment]
@zhouyangnk Yeah good luck with that :)

**Comment by philc** [comment]
We're investigating support for Firefox in #2425, and will consider Edge based on the lessons learned from shipping and maintaining support for Firefox.

**Comment by florianmonfort** [comment]
Any news on this?

**Comment by Kethku** [comment]
Ping. This is a deal breaker for edge for me. Would be awesome to have similar functionality in the edge browser.

**Comment by mrmr1993** [comment]
To get a bare bones port running, we'll need some stubs for the Chrome APIs. If someone running Windows can grab the latest stub files (`backgroundScriptsAPIBridge.js` and `contentScriptsAPIBridge.js`) following the instructions from [here](https://docs.microsoft.com/en-us/microsoft-edge/extensions/guides/porting-chrome-extensions), it should be easy enough to add them to kick-start the process.

**Comment by Kethku** [comment]
I'll try to take a look at it tonight and update. Would the best strategy be to create a fork and pull request so we can iterate?

**Comment by mrmr1993** [comment]
That sounds like the best way to go, yes.

**Comment by mrmr1993** [comment]
I've put the latest version of the API compatability layer in [this gist](https://gist.github.com/mrmr1993/a7e17cce72e946260c827048d3a5bf6e).

**Comment by KonTy** [comment]
Any news here guys? Anything I can do to help? Edge has become such a good browser I see no reason to install Chrome anymore except for this plugin :(

**Comment by mrmr1993** [comment]
I've tested getting (very) basic support in [my branch here](https://github.com/mrmr1993/vimium/tree/experiments/edge) (tracking via PR mrmr1993/vimium#9). To test, check out that branch, build it ([instructions](https://github.com/philc/vimium/blob/master/CONTRIBUTING.md#installing-from-source)) and then [add to Edge](https://docs.microsoft.com/en-us/microsoft-edge/extensions/guides/adding-and-removing-extensions).

This isn't a priority for me (I don't run Windows or use Edge) so please anybody feel free to jump in and try to improve things. My tracking PR has a list of the obvious things that would benefit from fixes.

**Comment by mikicho** [comment]
ping

**Comment by KonTy** [comment]
Yeah I know someone just has to go do it. Funny thing is I'm liking Edge more and more it has matured a lot and a quite a bit faster then Chrome now but I'm really missing this plug in.

**Comment by KonTy** [comment]
I wonder does it change anything now the fact that Chromium Edge is out. Did anyone look into it? How easy is it to load Chrome addons in Chromium Edge now?

**Comment by gdh1995** [comment]
The Edge Insider based on Chromium can load Chrome extensions directly and perfectly.

**Comment by nguyendat** [comment]
Yayy!, it work on Microsoft Edge Chromium !

**Comment by sebastiancarlos** [comment]
should this issue be closed, considering that Edge is Chromium-based now?

**Comment by mikicho** [comment]
Yes

**Comment by philc** [comment]
Thanks. I'll close this in favor of #4211. I do intend to publish Vimium to the Edge Addons site.

**Comment by philc** [comment]
Cross posting here, even though I posted this on #4211:

Hey folks, Vimium is now on the [Edge Add-ons site](https://microsoftedge.microsoft.com/addons/detail/vimium/djmieaghokpkpjfbpelnlkfgfjapaopa). Please leave a review there to help encourage others to try it. Thanks!

**Comment by KonTy** [comment]
@philc congrats for reaching this awesome milestone :) Now we need to make it work on Firefox :)

**Comment by philc** [comment]
@KonTy It [does](https://addons.mozilla.org/en-GB/firefox/addon/vimium-ff/)!
